osquery/tables/system/posix/ssh_configs.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/posix/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/system_utils.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kUserSSHConfig = ".ssh/config";
+const std::string kSystemwideSSHConfig = "/etc/ssh/ssh_config";
+
+void genSSHConfig(const std::string& uid,
+                  const std::string& gid,
+                  const boost::filesystem::path& filepath,
+                  QueryData& results) {
+  std::string ssh_config_content;
+  if (!forensicReadFile(filepath, ssh_config_content).ok()) {
+    // Cannot read a specific ssh_config file.
+    return;
+  }
+  // the ssh_config file consists of a number of host or match
+  // blocks containing newline-separated options for each
+  // block; a block is defined as everything following a
+  // host or match keyword, until the next host or match
+  // keyword, else EOF
+  std::string block;
+  for (auto& line : split(ssh_config_content, "\n")) {
+    boost::trim(line);
+    boost::to_lower(line);
+    if (!line.empty() && line[0] != '#') {
```

Comment:
I think you could flip your logic here and return early:
```
if (line.empty || line[0] == '#') {
  return;
}
```

Reply:
I like that better, thanks. I have made the change.

I considered also moving the check before the `boost::to_lower(line);`, but I feel the cost is probably rather low, and that this might be more readable.